### ISSUE TITLE
Unknown command "version --short" for "helm" cause the Jenkins pipeline to fail

### ISSUE THREAD

**Issue by CharliePF** [jenkins-x/jx#1259]
Build fails after Docker image is created and pushed. I've attached part of the build log and also version information of the JX components.

[jenkins-build-log.txt](https://github.com/jenkins-x/jx/files/2183643/jenkins-build-log.txt)
<img width="1130" alt="screen shot 2018-07-11 at 09 26 58" src="https://user-images.githubusercontent.com/40855520/42559951-6bb83482-84ed-11e8-8032-69915d1707d3.png">

**Comment by jstrachan** [comment]
here's the log as embedded text btw: 
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:15 min
[INFO] Finished at: 2018-07-10T16:20:19Z
[INFO] ------------------------------------------------------------------------
[Pipeline] sh
[-platform_app-server_master-JFBOUNKIUK5ALQ5JGWD7LQ3GJ4JGQXBODYYIBCGXVXJKNFKKMUFA] Running shell script
++ cat VERSION
+ export VERSION=6.0.6
+ VERSION=6.0.6
+ skaffold build -f skaffold.yaml
Starting build...
Building [app-server]...
Sending build context to Docker daemon    527kB
Sending build context to Docker daemon     11MB
Sending build context to Docker daemon  22.95MB
Sending build context to Docker daemon  38.49MB
Sending build context to Docker daemon  50.66MB
Sending build context to Docker daemon  64.47MB
Sending build context to Docker daemon  76.01MB
Sending build context to Docker daemon  85.88MB
Sending build context to Docker daemon  99.17MB
Sending build context to Docker daemon  112.3MB
Sending build context to Docker daemon  126.2MB
Sending build context to Docker daemon  141.8MB
Sending build context to Docker daemon  156.8MB
Sending build context to Docker daemon  164.5MB
Sending build context to Docker daemon  177.2MB
Sending build context to Docker daemon  192.1MB
Sending build context to Docker daemon  208.1MB
Sending build context to Docker daemon  220.3MB
Sending build context to Docker daemon  234.1MB
Sending build context to Docker daemon  247.8MB
Sending build context to Docker daemon  262.2MB
Sending build context to Docker daemon  267.7MB
Sending build context to Docker daemon  278.7MB
Sending build context to Docker daemon  294.3MB
Sending build context to Docker daemon  308.2MB
Sending build context to Docker daemon  321.5MB
Sending build context to Docker daemon  327.5MB
Sending build context to Docker daemon  328.5MB

Step 1/16 : FROM openjdk:8-jdk-alpine
 ---> 1caad94162ef
Step 2/16 : ENV PORT 8080
 ---> Using cache
 ---> 3e6fbdd8ae6f
Step 3/16 : ENV CLASSPATH /opt/app-server
 ---> Using cache
 ---> 2f8cf3f69af3
Step 4/16 : EXPOSE 8080
 ---> Using cache
 ---> 06dd41070acc
Step 5/16 : COPY target/****-client /opt/app-server/****-client
 ---> Using cache
 ---> 12bdc8ad4191
Step 6/16 : COPY target/appassembler /opt/app-server/
 ---> Using cache
 ---> 4fe14f5b760b
Step 7/16 : COPY target/config /opt/app-server/config
 ---> 91ffb66cc505
Step 8/16 : COPY target/generated-resources/appassembler/booter-unix/bin /opt/app-server/bin
 ---> baac5a2b7b1c
Step 9/16 : COPY target/generated-resources/appassembler/booter-unix/etc/app-server.xml /opt/app-server/etc
 ---> 3d687a4d2e10
Step 10/16 : COPY target/generated-resources/appassembler/booter-windows/bin /opt/app-server/bin
 ---> cbbe81695f46
Step 11/16 : COPY target/generated-resources/appassembler/booter-windows/etc/app-server.xml /opt/app-server/etc
 ---> ce67daa2f180
Step 12/16 : COPY target/generated-resources/appassembler/jsw/app-server-service /opt/app-server/
 ---> 201ae78c19d9
Step 13/16 : COPY target/web-client /opt/app-server/web-client
 ---> 6fde9096222d
Step 14/16 : RUN mkdir /opt/app-server/forms && mkdir /opt/app-server/logs && mkdir /opt/app-server/modules
 ---> Running in 6575aed1f3b7
 ---> d1af6827badc
Step 15/16 : WORKDIR /opt/app-server
 ---> 6019370ac4ac
Step 16/16 : ENTRYPOINT sh /opt/app-server/bin/app-server
 ---> Running in a76d8d95ebd4
 ---> 1b56539b6c9f
Successfully built 1b56539b6c9f
Successfully tagged 10.27.250.50:5000/charlie-m/app-server:6.0.6
The push refers to a repository [10.27.250.50:5000/charlie-m/app-server]
bd379b033148: Preparing
e7a9839306d0: Preparing
a0eacb12d0c7: Preparing
accad5c0b98e: Preparing
b64e04b81457: Preparing
6346c277a27d: Preparing
6d70e7253963: Preparing
7ed136c1e497: Preparing
b2494c007840: Preparing
c0939b1c0ba5: Preparing
138726990cf7: Preparing
5b40c5fb5220: Preparing
717b092b8c86: Preparing
6346c277a27d: Waiting
6d70e7253963: Waiting
7ed136c1e497: Waiting
b2494c007840: Waiting
c0939b1c0ba5: Waiting
138726990cf7: Waiting
5b40c5fb5220: Waiting
717b092b8c86: Waiting
b64e04b81457: Pushed
accad5c0b98e: Pushed
bd379b033148: Pushed
6346c277a27d: Pushed
b2494c007840: Layer already exists
6d70e7253963: Pushed
c0939b1c0ba5: Layer already exists
7ed136c1e497: Pushed
138726990cf7: Layer already exists
717b092b8c86: Layer already exists
5b40c5fb5220: Layer already exists
e7a9839306d0: Pushed
a0eacb12d0c7: Pushed
6.0.6: digest: sha256:d1dd0494ac857b292552f6588144d6ed0062ed26835238842370e9a9f0dd7340 size: 3041
Build complete in 16.790500921s
app-server -> 10.27.250.50:5000/charlie-m/app-server:6.0.6
[Pipeline] sh
[-platform_app-server_master-JFBOUNKIUK5ALQ5JGWD7LQ3GJ4JGQXBODYYIBCGXVXJKNFKKMUFA] Running shell script
++ cat VERSION
+ jx step post build --image 10.27.250.50:5000/charlie-m/app-server:6.0.6
no CVE provider running in the current jx namespace so skip adding image to be analysed[Pipeline] }
[Pipeline] // container
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Promote to Environments)
[Pipeline] dir
Running in /home/jenkins/workspace/-platform_app-server_master-JFBOUNKIUK5ALQ5JGWD7LQ3GJ4JGQXBODYYIBCGXVXJKNFKKMUFA/charts/app-server
[Pipeline] {
[Pipeline] container
[Pipeline] {
[Pipeline] sh
[app-server] Running shell script
++ cat ../../VERSION
+ jx step changelog --version v6.0.6
Using batch mode as inside a pipeline
Generating change log from git ref 10cb88b4cc15bfa72191284c0ca57a8fa0321bd5 => 71932eda6534542b3e89eae4f5ee4c2ce2d45e39
Unable to find user: jenkins-x.googlegroups.com -- users.jenkins.io "jenkins-x.googlegroups.com" not found
Unable to find user: jenkins-x.googlegroups.com -- users.jenkins.io "jenkins-x.googlegroups.com" not found
Finding issues in commit messages using git format
Failed to lookup issue 12 in issue tracker https://bitbucket.org/pf-platform/app-server due to Status: 404 Not Found, Body: {"type": "error", "error": {"message": "Repository has no issue tracker."}}
Bitbucket Cloud doesn't support releases
Updated the release information at https://bitbucket.org/pf-platform/app-server/releases/tag/v6.0.6
generated: /home/jenkins/workspace/-platform_app-server_master-JFBOUNKIUK5ALQ5JGWD7LQ3GJ4JGQXBODYYIBCGXVXJKNFKKMUFA/charts/app-server/templates/release.yaml
[Pipeline] sh
[app-server] Running shell script
+ jx step helm release
No $CHART_REPOSITORY defined so using the default value of: http://jenkins-x-chartmuseum:8080
error: failed to build dependencies for chart from directory '.': failed to read the helm version: failed to run 'helm version --short' command in directory '.', output: 'Error: unknown command "version --short" for "helm"
Run 'helm --help' for usage.': exit status 1
[Pipeline] }
[Pipeline] // container
[Pipeline] }
[Pipeline] // dir
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
[Pipeline] input
Pipeline failed. 
We will keep the build pod around to help you diagnose any failures. 

Select Proceed or Abort to terminate the build pod
Proceed or Abort
Error when executing failure post condition:
org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
```

**Comment by jstrachan** [comment]
@CharliePF whats the output of:
```
jx version
helm version
```

**Comment by CharliePF** [comment]
Last login: Wed Jul 11 09:18:18 on ttys003
~ ❯❯❯ jx version
Failed to get helm version: failed to run 'helm version --short' command in directory '', output: 'Error: unknown command "version --short" for "helm"
Run 'helm --help' for usage.': exit status 1
NAME               VERSION
jx                 1.3.77
jenkins x platform 0.0.1774
kubernetes cluster v1.9.7-gke.3
kubectl            v1.9.7
git                git version 2.17.0 (Apple Git-106)

A new jx version is available: 1.3.79
? Would you like to upgrade to the new jx version? No

~ ❯❯❯ helm version
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
~ ❯❯❯

**Comment by jstrachan** [comment]
odd - whats the output of `helm help version`

**Comment by jstrachan** [comment]
aha found it: https://github.com/jenkins-x/jx/pull/1261 - should have a release soon

**Comment by CharliePF** [comment]
Quick turnaround. Thanks.

**Comment by jstrachan** [comment]
its available now in `1.3.81` or later: https://github.com/jenkins-x/jx/releases
